parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -18,13 +18,29 @@
  */
 package org.apache.parquet.column.statistics;
 
+import org.apache.parquet.column.Comparators;
 import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
 
 public class BinaryStatistics extends Statistics<Binary> {
 
+  private final Comparators.BinaryComparator comparator;
   private Binary max;
   private Binary min;
 
+  /**
+   * @deprecated Use {@link Statistics#getStatsBasedOnType(PrimitiveType.PrimitiveTypeName, OriginalType)} instead
+   */
+  @Deprecated
+  public BinaryStatistics() {
+    this(PrimitiveType.PrimitiveTypeName.BINARY, null);
```

Comment:
From the changes to parquet-cli, it looks like you intend for `BinaryStatistics` to be used for all types. If that's the case, then we should remove the other statistics implementations. I think the only class considered public is the `Statistics<T>` interface.